File: introduction/11614/c++/main.cpp
#include <iostream>
#include <cmath>

using namespace std;

int CalculatingRows(int64_t warriors) {
    if (warriors > 0) {
        int auxiliary = sqrt((8 * warriors) + 1);
        return (auxiliary - 1) / 2;
    } else {
        return 0;
    }
}

int main() {
    int64_t n;
    cin >> n;
    
    if (n >= 0 && n < pow(10, 18)) {
        int64_t warriors;
        for (int64_t i = 0; i < n; i++) {
            cin >> warriors;
            cout << CalculatingRows(warriors) << endl;
        }
    }
    
    return 0;
}

File: introduction/11614/go/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int64
	_, erro := fmt.Scanf("%d", &n)
	if erro == nil && n >= 0 && n < int64(math.Pow10(18)) {

		var warriors int64
		var i int64
		for i = 0; i < n; i += 1 {
			_, erro := fmt.Scanf("%d", &warriors)
			if erro == nil {
				fmt.Printf("%d\n", CalculatingRows(warriors))
			}
		}
	}
}

func CalculatingRows(warriors int64) (result int) {
	if warriors > 0 {
		auxiliary := int(math.Sqrt(float64((8 * warriors) + 1)))
		result = (auxiliary - 1) / 2
		return result
	} else {
		return 0
	}
}
